gruntfile.js

```diff
@@ -4,328 +4,373 @@
+      }
+    },
+    mocha_istanbul: {
+      coverage: {
+        src: testAssets.tests.server,
+        options: {
+          print: 'detail',
+          coverage: true,
+          require: 'test.js',
+          coverageFolder: 'coverage',
+          reportFormats: ['cobertura','lcovonly'],
+          check: {
+            lines: 40,
+            statements: 40
+          }
+        }
+      }
+    },
+    karma: {
+      unit: {
+        configFile: 'karma.conf.js'
+      }
+    },
+    protractor: {
+      options: {
+        configFile: 'protractor.conf.js',
+        keepAlive: true,
+        noColor: false
+      },
+      e2e: {
+        options: {
+          args: {} // Target-specific arguments
+        }
+      }
+    },
+    copy: {
+      localConfig: {
+        src: 'config/env/local.example.js',
+        dest: 'config/env/local.js',
+        filter: function () {
+          return !fs.existsSync('config/env/local.js');
+        }
+      },
+      tinyjson: {
+        expand: true,
+        cwd: 'node_modules/tiny-jsonrpc',
+        src: '*',
+        dest: 'node_modules/spooky/node_modules/tiny-jsonrpc',
+      }
+    }
+  });
+
+  grunt.event.on('coverage', function(lcovFileContents, done) {
+    require('coveralls').handleInput(lcovFileContents, function(err) {
+      if (err) {
+        return done(err);
+      }
+      done();
+    });
+  });
+
+  // Load NPM tasks
+  require('load-grunt-tasks')(grunt);
+
+  //grunt.loadNpmTasks('grunt-sass');
+  grunt.loadNpmTasks('grunt-ng-constant');
+
+  grunt.task.registerTask('buildconstants', 'Builds all the environment information and bakes it into a conf.js file.', function () {
+    grunt.task.run('ngconstant');
+  });
+
+  // Make sure upload directory exists
+  grunt.task.registerTask('mkdir:upload', 'Task that makes sure upload directory exists.', function () {
+    // Get the callback
+    var done = this.async();
+
+    grunt.file.mkdir(path.normalize(__dirname + '/modules/users/client/img/profile/uploads'));
+
+    done();
+  });
+
+  // Connect to the MongoDB instance and load the models
+  grunt.task.registerTask('mongoose', 'Task that connects to the MongoDB instance and loads the application models.', function () {
+    // Get the callback
+    var done = this.async();
+
+    // Use mongoose configuration
+    var mongoose = require('./config/lib/mongoose.js');
+
+    // Connect to database
+    mongoose.connect(function () {
+      done();
+    });
+  });
+
+  var server_proc;
+
+  grunt.task.registerTask('server', 'Starting server...', function() {
+    var done = this.async();
+
+    server_proc = childProcess.spawn('node', ['server.js'],
+      {
+        env: process.env,
+        detached: true,
+        shell: false,
+        stdio: 'inherit'
+      });
+
+    done();
+  });
+
+  grunt.task.registerTask('functional', 'Starting functional tests...', function () {
+    var done = this.async();
+
+    //'-DchromeTest.single=AddEditProjectSpec', --before chromeTest
```

Comment:
Is this comment necessary?

Reply:
convenience commment - I never rememmber how to do this. Not stricly necessary, but I found it helpful.